### PR TITLE
Front-end Rewrite

### PR DESCRIPTION
**Pull request by okunamayanad**
A fresh front-end

### CODE REVIEW

File: src/lib/AuthButtons.tsx [hunk 1]
```diff
@@ -0,0 +1,221 @@
...
+          <DrawerTitle>just read the instructions next time lol</DrawerTitle>
+          <DrawerDescription>
+            if you were to read the instructions, you would know that you need
+            to create an account in-game first.
+          </DrawerDescription>
+        </DrawerHeader>
+        <DrawerFooter>
+          <DrawerClose />
+        </DrawerFooter>
+      </DrawerContent>
```
Comment (tkbstudios) [inline src/lib/AuthButtons.tsx:79]: Update this to redirect to https://ccbank.tkbstudios.com/api/v1/webauth

File: src/lib/AuthButtons.tsx [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+              }
+            : {})}
+        >
+          <Button variant={"outline"}>Logout</Button>
+        </DialogTrigger>
+        <DialogContent>
+          <DialogHeader>
+            <DialogTitle>Logout</DialogTitle>
+            <DialogDescription>
+              Are you sure logging out? You can always log back in.
```
Comment (tkbstudios) [inline src/lib/AuthButtons.tsx:155]: "Are you sure you want to log out? You can always log back in." I think this is a more proper way of telling it

File: src/lib/AuthButtons.tsx [hunk 3]
```diff
@@ -0,0 +1,221 @@
...
+            }
+          : {})}
+      >
+        <Button variant={"outline"}>Logout</Button>
+      </DrawerTrigger>
+      <DrawerContent>
+        <DrawerHeader>
+          <DrawerTitle>Logout</DrawerTitle>
+          <DrawerDescription>
+            Are you sure logging out? You can always log back in.
```
Comment (tkbstudios) [inline src/lib/AuthButtons.tsx:195]: "Are you sure you want to log out? You can always log back in." I think this is a more proper way of telling it

File: src/lib/LoginForm.tsx [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+      toast.error(data.message);
+      setProcessing(false);
+      return;
+    }
+
+    toast.success("Logged in successfully!");
+    setCookie("session_token", data.session_token, { secure: true });
+
+    // redirect to dashboard
+    window.location.href = "/dashboard/15/1";
```
Comment (tkbstudios) [inline src/lib/LoginForm.tsx:78]: /dashboard would be a better way of using it.

**Review by tkbstudios — CHANGES_REQUESTED** [review]
I left some comments for changes

File: src/lib/AuthButtons.tsx (hunk 2 above)
Comment (okunamayanad) [reply src/lib/AuthButtons.tsx]: Changed as requested

File: src/lib/LoginForm.tsx (hunk 4 above)
Comment (okunamayanad) [reply src/lib/LoginForm.tsx:78]: Current focus is to get this off of this PR so contributions may continue. Will implement on another PR.

Comment (tkbstudios) [reply src/lib/LoginForm.tsx:78]: alright 👍

File: src/app/page.tsx [hunk 5]
```diff
@@ -10,13 +10,26 @@ import { FormDescription } from "@/components/ui/form";
...
 export default function Home() {
+  const searchParams = useSearchParams();
+
   const [isLogged, setIsLogged] = React.useState(false);
   const [isLoaded, setIsLoaded] = React.useState(false);
   const [isLoginDialogOpen, setLoginDialogOpen] = React.useState(false);
   const [isProcessing, setProcessing] = React.useState(false);
 
+  const isRegistered =
+    searchParams.get("action") == ("registered" || "loggedIn");
```
Comment (tkbstudios) [inline src/app/page.tsx:24]: it's loggedin instead of loggedIn

**Review by tkbstudios — CHANGES_REQUESTED** [review]

**Pull request merged**